File: backend/src/main/java/com/calonuria/backend/features/journal/dto/BaseJournalResponseDTO.java
package com.calonuria.backend.features.journal.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Clase base abstracta para DTOs de respuesta de diario de lectura.
 * Contiene los campos comunes compartidos entre libros, mangas y fanfictions.
 */
@Data
public abstract class BaseJournalResponseDTO {

    private Long id;
    private Long userId;
    private String status;
    private Integer rating;
    private Integer tearDrops;
    private Integer spiceFlames;
    private String personalNotes;
    
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;
    
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;
    
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", timezone = "UTC")
    private LocalDateTime updatedAt;

    private Boolean rereading;
    private String readingFormat;
    private String ownership;
    private String loanedTo;
}

File: frontend/between_pages/lib/features/journal/domain/responses/base_journal_response_dto.dart
abstract class BaseJournalResponseDTO {
  final int id;
  final int userId;
  final String status;
  final String? personalNotes;
  final String? startDate;
  final String? endDate;
  final int? rating;
  final int? tearDrops;
  final int? spiceFlames;
  final bool? rereading;
  final String? ownership;
  final String? readingFormat;
  final String? loanedTo;
  final String? updatedAt;

  // Campos que las clases hijas deben implementar
  final String? coverUrl;
  final String? title;
  final String? author;

  const BaseJournalResponseDTO({
    required this.id,
    required this.userId,
    required this.status,
    this.personalNotes,
    this.startDate,
    this.endDate,
    this.rating,
    this.tearDrops,
    this.spiceFlames,
    this.rereading,
    this.ownership,
    this.readingFormat,
    this.loanedTo,
    this.updatedAt,
    this.coverUrl,
    this.title,
    this.author,
  });
}
